app/assets/javascripts/actions/server_actions.js

```diff
@@ -13,6 +13,14 @@ const ServerActions = Flux.createActions({
       .catch(resp => ({ actionType: 'API_FAIL', data: resp }));
   },
 
+  fetchMoreRevisions(courseId, limit) {
+    const actionType = `RECEIVE_MORE_REVISIONS`;
```

Comment:
I think the disconnection with rendering the revisions is here. The action type determines which stores will receive and process the fetched data, and there is no store that handles `RECEIVE_MORE_REVISIONS`. If you change it to `RECEIVE_REVISIONS`, that may fix it.

I would suggest, rather than 'fetchMoreRevsions`, you just call this `fetchRevisions`, and use the same action (just with the default limit) in componentWillMount.

Reply:
What do you exactly mean? 'fetchRevisions' already exist and I think it does retrieve different endpoint (revisions for each student). I have tried and didn't work, but I will check it further. 
I added though 'RECEIVE_REVISIONS' and indeed works. Thanks.

Reply:
You're right, I misread the code. Currently, the default fetch revisions for the course is going through API.fetch(courseId, model). So the way to avoid using two different methods for basically the same thing, you could call the new method `fetchCourseRevisions`, and use it instead of general-purpose fetch method for both the default and 'more' data flows.